ED/protocol.html

```diff
@@ -757,7 +757,7 @@ <h3 property="schema:name">HTTP Client</h3>
 
                   <p about="" id="client-authentication-different-credentials" rel="spec:requirement" resource="#client-authentication-different-credentials"><span property="spec:statement">When a <span rel="spec:requirementSubject" resource="#Client">client</span> receives a response with a <code>403</code> or <code>404</code> status code, the client <span rel="spec:requirementLevel" resource="spec:MAY">MAY</span> repeat the request with different credentials.</span></p>
 
-                  <p about="" id="client-content-type-includes" rel="spec:requirement" resource="#client-content-type"><span property="spec:statement"><span rel="spec:requirementSubject" resource="#Client">Clients</span> <span rel="spec:requirementLevel" resource="spec:MUST">MUST</span> use the <code>Content-Type</code> HTTP header in <code>PUT</code>, <code>POST</code>, and <code>PATCH</code> requests [<cite><a class="bibref" href="#bib-rfc9110">RFC9110</a></cite>].</span> [<a href="https://github.com/solid/specification/issues/70#issuecomment-547924171" rel="cito:citesAsSourceDocument">Source</a>]</p>
+                  <p about="" id="client-content-type-includes" rel="spec:requirement" resource="#client-content-type"><span property="spec:statement"><span rel="spec:requirementSubject" resource="#Client">Clients</span> <span rel="spec:requirementLevel" resource="spec:MUST">MUST</span> use the <code>Content-Type</code> HTTP header in <code>PUT</code>, <code>POST</code>, and <code>PATCH</code> requests including content [<cite><a class="bibref" href="#bib-rfc9110">RFC9110</a></cite>].</span> [<a href="https://github.com/solid/specification/issues/70#issuecomment-547924171" rel="cito:citesAsSourceDocument">Source</a>]</p>
```

Comment:
Suggest making the language more RFC9110 like and also include correctness of media-type as a condition:

"When sending a PUT, POST and PATCH requests with a message containing content, a client MUST generate a as part of the message the Content-Type header field with the media type of the content".

Reply:
I don't think this (or any) requirement should be more RFC9110, not that there is anything wrong with that. The language in this document is aimed to be internally consistent first and foremost, and the conformance clause template that's in use is along the lines what we see right now. There are variations of course throughout the spec.

I have changed "including" to "containing" and added "field" to the text as per your suggestion. Thanks.

Reply:
This definitely is an extra requirement on the client. My reasoning being that if we get rid of sniffing completely, and the server does not (have to) have the possibility of checking the provided content type which it now MUST process (as opposed to SHOULD in RFC9110), the burden is entirely on the client to provide the right content-type and not expect the server to recover from the wrong one. This has to be made explicit.

Reply:
Having mention this being an extra requirement, I still think it belongs here because it is intimately tied to the change from SHOULD to MUST.

Reply:
The general point is that the sender of the message (i.e., both the server and the client) needs to know what they are sending. The context of the sniffing in RFC and in Solid Protocol is limiting or eliminating the MIME sniffing cases. When the client includes the Content-Type header field and a field value, the server is not forced into a situation (code path) to sniff or reject outright.

Surely the server already has, and needs to have, all the rights to inspect, process, and handle errors, as it deems necessary. That's not the same as "sniffing".

Reply:
Sure, server can reject any request it does not like. The point here is that client due to the change now has a mandatory obligation to ensure media-type correctness, which was not the case before 

Again lets look at the RFC9110 8.3-5,
> A sender that generates a message containing content SHOULD generate a Content-Type header field in that message unless the intended media type of the enclosed representation is unknown to the sender.

When we change the SHOULD in RFC9110 to a MUST in Solid specification, the exception provided to the client of not knowing the media type of the content no longer applies. The client hence has that additional burden over that of HTTP clients of providing the correct media-type which we should make explicit.

Reply:
Again, correct use of HTTP is already required for interoperability:
* https://solidproject.org/ED/protocol#server-http
* https://solidproject.org/ED/protocol#client-http

And Conformance to HTTP is given: https://www.rfc-editor.org/rfc/rfc9110#name-conformance

Changing the requirement level from SHOULD to MUST does not constitute a harder, better, faster, stronger adherence to the definition of `Content-Type` header field and field value, or "media-type correctness" (or for any other pat of HTTP.) It is a given.